Sources/Lumina/UI/LuminaButton.swift

```diff
@@ -0,0 +1,189 @@
+        self.frame = CGRect(origin: CGPoint(x: UIScreen.main.bounds.maxX - 50, y: 10), size: CGSize(width: self.squareSystemButtonWidth, height: self.squareSystemButtonHeight))
+      case .cancel:
+        self.image = UIImage(systemName: "xmark.circle.fill", withConfiguration: symbolConfig)?.withTintColor(.white, renderingMode: .alwaysOriginal)
+        self.frame = CGRect(origin: CGPoint(x: UIScreen.main.bounds.maxX - 50, y: 10), size: CGSize(width: self.squareSystemButtonWidth, height: self.squareSystemButtonHeight))
+      case .shutter:
+        self.backgroundColor = UIColor.normalState
+        var minY = UIScreen.main.bounds.maxY
+        if #available(iOS 11, *) {
+          minY = self.safeAreaLayoutGuide.layoutFrame.maxY
+        }
+        minY -=  80
+        self.frame = CGRect(origin: CGPoint(x: UIScreen.main.bounds.midX - 35, y: minY), size: CGSize(width: self.shutterButtonDimension, height: self.shutterButtonDimension))
+        self.layer.cornerRadius = CGFloat(self.shutterButtonDimension / 2)
+        self.layer.borderWidth = 3
+        self.layer.borderColor = UIColor.borderNormalState
+      default:
+        break
+    }
+  }
+
+  func startRecordingVideo() {
+    if style == .shutter {
+      DispatchQueue.main.async {
+        UIView.animate(withDuration: 0.1, animations: {
+          self.backgroundColor = UIColor.recordingState
+          self.layer.borderColor = UIColor.borderRecordingState
+          self.transform = CGAffineTransform(scaleX: 1.3, y: 1.3)
+        })
+      }
+    }
+  }
+
+  func stopRecordingVideo() {
+    if style == .shutter {
+      DispatchQueue.main.async {
+        UIView.animate(withDuration: 0.1, animations: {
+          self.backgroundColor = UIColor.normalState
+          self.layer.borderColor = UIColor.borderNormalState
+          self.transform = CGAffineTransform(scaleX: 1.0, y: 1.0)
+        })
+      }
+    }
+  }
+
+  func takePhoto() {
+    if style == .shutter {
+      DispatchQueue.main.async {
+        UIView.animate(withDuration: 0.1, animations: {
+          self.backgroundColor = UIColor.takePhotoState
+          self.layer.borderColor = UIColor.borderTakePhotoState
+        }, completion: { _ in
+          UIView.animate(withDuration: 0.1, animations: {
+            self.backgroundColor = UIColor.normalState
+            self.layer.borderColor = UIColor.borderNormalState
+          })
+        })
+      }
+    }
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+  }
+}
+
+extension LuminaButton {
+  func updateTorchIcon(to state: SystemButtonType.FlashState) {
+    guard let style = self.style, style == .torch else {
+      return
+    }
+    switch state {
+      case .on:
+        self.image = UIImage(systemName: "bolt.fill", withConfiguration: symbolConfig)?.withTintColor(.white, renderingMode: .alwaysOriginal)
+        LuminaLogger.debug(message: "torch icon updated to on")
+      case .off:
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dokun1/Lumina/blob/f3c4aa8d5c66dd9d2660cdba8e658b8902c9752f/Sources/Lumina/UI/LuminaButton.swift#L155"></span>Case statements should vertically align with their enclosing switch statement.<br />
<code>switch_case_alignment</code> <code>LuminaButton.swift:155</code></td>
    </tr>
  </tbody>
</table>